script.js

```diff
@@ -1,4 +1,300 @@
+    var outputText = `LEADERBOARD <br><br> Player 1 - Score: ${player1Score} <br> Player 2 - Score: ${player2Score}`;
+  } else {
+    var outputText = `LEADERBOARD <br><br> Player 2 - Score: ${player2Score} <br> Player 1 - Score: ${player1Score}`;
+  }
+
+  outputText =
+    outputWord +
+    `You chose '${userInputs[inputIndex]} combined number mode'. <br><br> The winner of this round: ${winnerText}.<br><br><br>` +
+    outputText +
+    `<br><br><br>OTHER STATISTICS <br><br> Player 1 - Current Number: ${player1NumberList[numberIndex]} <br> Player 1 - All Numbers: ${player1NumberList} <br><br> Player 2 - Current Number: ${player2NumberList[numberIndex]} <br> Player 2 - All Numbers: ${player2NumberList} <br><br><br> Play again by inputting either "highest" to play the 'highest combined number mode' or "lowest" to play the 'lowest combined number mode'`;
+
+  // Restart the game or play again
+  gameMode = GAME_MODE_CHOOSE_HIGHEST_OR_LOWEST;
+
+  // Update dice index count
+  diceIndex += 2;
+  inputIndex += 1;
+  numberIndex += 1;
+  numberOfDiceIndex += 1;
+
+  // Redefine values of number to 0
+  player1Number = 0;
+  player2Number = 0;
+
+  return outputText;
+};
+
+// Decide who the winner is (i.e. whose number is higher)
+var decideWinner = function () {
+  if (player1Number > player2Number) {
+    return "Player 1";
+  } else {
+    return "Player 2";
+  }
+};
+
+// Decide who the temporary leader is by score
+var decideWinnerByScore = function () {
+  if (player1Score > player2Score) {
+    return "Player 1";
+  } else {
+    return "Player 2";
+  }
+};
+
+// Decide who the winner is (i.e. whose number is lowest)
+var decideWinnerByLowestNumber = function () {
+  if (player1Number < player2Number) {
+    return "Player 1";
+  } else {
+    return "Player 2";
+  }
+};
+
+// Generate score - running sum of all numbers that player has generated so far
+var generateScore = function (array) {
+  var score = 0;
+  var index = 0;
+  while (index < array.length) {
+    score = score + array[index];
+    index += 1;
+  }
+  return score;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  userInput = input;
+  var myOutputValue = "hello";
+
+  // Early return for gameMode = "choose highest or lowest"
+  if (gameMode == GAME_MODE_CHOOSE_HIGHEST_OR_LOWEST) {
+    myOutputValue = chooseHighestModeOrLowestMode();
+    return myOutputValue;
+  }
+
+  // Early return for gameMode = "choose number of dice"
+  if (gameMode == GAME_MODE_CHOOSE_NUMBER_OF_DICE) {
+    myOutputValue = chooseNumberOfDice();
+    return myOutputValue;
+  }
+
+  if (gameMode == GAME_MODE_PLAYER_ONE_ROLL) {
+    myOutputValue = playerOneRoll();
+  } else if (gameMode == GAME_MODE_PLAYER_ONE_CHOOSE) {
+    myOutputValue = playerOneChoose();
+  } else if (gameMode == GAME_MODE_PLAYER_TWO_ROLL) {
+    myOutputValue = playerTwoRoll();
+  } else if (gameMode == GAME_MODE_PLAYER_TWO_CHOOSE) {
+    myOutputValue = playerTwoChoose();
```

Comment:
great job extracting out the logic part of your code and placing it as helper functions outside your main function